### PR TITLE
Add separator option to overflow icon menu

### PR DESCRIPTION
**Pull request by spacegaier**
## Proposed change

Added the option to insert a divider / separator to the `ha-icon-overflow-menu` which ensures that the menu can now look the same as the overflow menus in the automation, script, scene editor.

Non-narrow variant now also covered thanks to Bram :)

image

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Review by bramkragten — DISMISSED** [review]

File: src/components/ha-icon-overflow-menu.ts [hunk 1]
```diff
@@ -46,30 +47,32 @@ export class HaIconOverflowMenu extends LitElement {
...
+                    </mwc-list-item> `
               )}
             </ha-button-menu>`
         : html`
             <!-- Icon representation for big screens -->
             ${this.items.map((item) =>
               item.narrowOnly
                 ? ""
+                : item.divider
+                ? html`<li divider role="separator"></li>`
```
Comment (bramkragten) [inline src/components/ha-icon-overflow-menu.ts:75]: Does this work? As it is not a list or something...

Comment (spacegaier) [reply src/components/ha-icon-overflow-menu.ts:75]: I had tested via Chome dev tools both in desktop mode and in mobile mode and in both scenarios I see the separator in the menu.

Comment (bramkragten) [reply src/components/ha-icon-overflow-menu.ts:75]: you have to set narrow explicitly, only backup panel uses it now I think

Comment (spacegaier) [reply src/components/ha-icon-overflow-menu.ts:75]: It is already in narrow mode, so the non-narrow one is actually the issue: https://github.com/spacegaier/frontend/blob/ec257710ffc4298ba9a2f270b39b9900ee0a1d31/src/panels/config/automation/ha-automation-picker.ts#L192

Will push a fix in a minute.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**